apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryResults.cs

```diff
@@ -192,5 +260,58 @@ public BigQueryResults ThrowOnAnyError()
             }
             return this;
         }
+
+        private sealed class AsyncRowEnumerable : IAsyncEnumerable<BigQueryRow>
+        {
+            private readonly BigQueryResults _parent;
+
+            public AsyncRowEnumerable(BigQueryResults parent) => _parent = parent;
+
+            public IAsyncEnumerator<BigQueryRow> GetEnumerator() => new AsyncRowEnumerator(_parent);
+        }
+
+        private sealed class AsyncRowEnumerator : IAsyncEnumerator<BigQueryRow>
+        {
+            private readonly GetQueryResultsOptions _options;
+            private readonly BigQueryResults _parent;
+            private IEnumerator<BigQueryRow> _underlyingIterator;
+
+            public AsyncRowEnumerator(BigQueryResults parent)
+            {
+                _parent = parent;
+                _options = parent._options?.Clone() ?? new GetQueryResultsOptions();
+                _options.StartIndex = null;
+                _options.PageToken = parent._response.PageToken;
+                _underlyingIterator = parent.ResponseRows.GetEnumerator();
+            }
+
+            public BigQueryRow Current => _underlyingIterator.Current;
+
+            public async Task<bool> MoveNext(CancellationToken cancellationToken)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Should this be inside the `while` loop below?

Reply:
We at least need one call *outside* the loop, so that it can still be cancelled even if we're iterating over the content of an existing response. Within the loop, I don't *think* we need one as I'd expect the cancellation token usage in the RPC to be enough.

Reply:
Ah yes, I had missed that `cancellationToken` is passed into `GetRawQueryResultsAsync(...)`.